waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -848,6 +853,16 @@
                                        :requests {:cpu (str (:cpu resources)) :memory (str (:mem resources) "Mi")}}}]
       envoy-container)))
 
+(def ^:const service-ports-index 0)
+(def ^:const proxied-ports-index 1)
+
+(defn get-port-range
+  "0th port in a range of up to 10 for a given service-id's hash.
+   Note that each container is limited to a contiguous 100 ranges of 10 ports,
+   therefore the return values are only unique for range-index [0, 99]."
+  [service-id-hash range-index base-port]
+  (-> service-id-hash (+ range-index) (mod 100) (* 10) (+ base-port)))
```

Comment:
Nitpick: we should extract the `10` in a constant, preferably use the same value in port value constraints.

Reply:
Since I'm not going to audit all of the uses of `10` right now, I opened an issue for this: https://github.com/twosigma/waiter/issues/1204